arroyo/processing/strategies/streaming/transform.py

```diff
@@ -354,27 +391,31 @@ def __submit_batch(self) -> None:
         self.__batch_builder = None
 
     def __check_for_results(self, timeout: Optional[float] = None) -> None:
-        input_batch, result = self.__results[0]
+        input_batch, async_result = self.__results[0]
 
         # If this call is being made in a context where it is intended to be
         # nonblocking, checking if the result is ready (rather than trying to
         # retrieve the result itself) avoids costly synchronization.
-        if timeout == 0 and not result.ready():
+        if timeout == 0 and not async_result.ready():
             # ``multiprocessing.TimeoutError`` (rather than builtin
             # ``TimeoutError``) maintains consistency with ``AsyncResult.get``.
             raise multiprocessing.TimeoutError()
 
-        i, output_batch = result.get(timeout=timeout)
+        result = async_result.get(timeout=timeout)
+        bad_messages = result.bad_messages
 
         # TODO: This does not handle rejections from the next step!
-        for message in output_batch:
-            self.__next_step.poll()
-            self.__next_step.submit(message)
+        for message in result.output_batch:
+            try:
+                self.__next_step.poll()
+                self.__next_step.submit(message)
+            except InvalidMessage as e:
```

Comment:
I don't think this is correct. the `poll` could be called on a batching step itself that can return multiple bad messages.

Reply:
Added another except for this just below, but I'll refactor following this:
https://github.com/getsentry/arroyo/pull/61#discussion_r855610674